src/main/java/it/unitn/ds1/Replica.java

```diff
@@ -3,29 +3,41 @@
 import java.io.Serializable;
 import java.util.AbstractMap;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Random;
 import java.util.Set;
+import java.util.concurrent.TimeUnit;
 
 import akka.actor.AbstractActor;
 import akka.actor.ActorRef;
+import akka.actor.Cancellable;
 import akka.actor.Props;
 import it.unitn.ds1.models.CrashMsg;
 import it.unitn.ds1.models.CrashResponseMsg;
-import it.unitn.ds1.models.HeartbeatMsg;
 import it.unitn.ds1.models.JoinGroupMsg;
 import it.unitn.ds1.models.ReadMsg;
 import it.unitn.ds1.models.ReadOkMsg;
+import it.unitn.ds1.models.StartMsg;
 import it.unitn.ds1.models.UpdateRequestMsg;
 import it.unitn.ds1.models.WriteAckMsg;
 import it.unitn.ds1.models.WriteMsg;
 import it.unitn.ds1.models.WriteOkMsg;
+import it.unitn.ds1.models.crash_detection.HearbeatReceivedMsg;
+import it.unitn.ds1.models.crash_detection.HeartbeatMsg;
+import it.unitn.ds1.models.crash_detection.WriteMsgReceivedMsg;
+import it.unitn.ds1.models.crash_detection.WriteOkReceivedMsg;
+import scala.concurrent.duration.Duration;
 
 public class Replica extends AbstractActor {
     static final int CRASH_CHANCES = 100;
+    static final int DELAY = 100;
```

Comment:
At this point, we could put all the constants in Utils, so that we don't use too many lines here.